CMakeLists.txt

```diff
@@ -540,120 +454,59 @@ if(CCF_RAFT_TRACING)
 endif()
 
 # Build common library for CCF enclaves
-add_custom_target(ccf ALL)
-
 set(CCF_IMPL_SOURCE
     ${CCF_DIR}/src/enclave/main.cpp ${CCF_DIR}/src/enclave/enclave_time.cpp
     ${CCF_DIR}/src/enclave/thread_local.cpp ${CCF_DIR}/src/node/quote.cpp
     ${CCF_DIR}/src/node/uvm_endorsements.cpp
 )
 
-# Same as virtual for the time being but will diverge soon
-if(COMPILE_TARGET STREQUAL "snp")
-
-  # SNP version
-  add_library(ccf.snp STATIC ${CCF_IMPL_SOURCE})
-
-  target_compile_definitions(
-    ccf.snp PUBLIC INSIDE_ENCLAVE VIRTUAL_ENCLAVE
-                   _LIBCPP_HAS_THREAD_API_PTHREAD PLATFORM_SNP
-  )
-
-  target_compile_options(ccf.snp PUBLIC ${COMPILE_LIBCXX})
-  add_warning_checks(ccf.snp)
-  target_compile_definitions(ccf.snp PRIVATE CCF_LOGGER_NO_DEPRECATE)
-
-  target_include_directories(
-    ccf.snp SYSTEM
-    PUBLIC
-      $<BUILD_INTERFACE:${CCF_GENERATED_DIR}>
-      $<INSTALL_INTERFACE:include/ccf/> #< This contains the private headers
-                                        #< which are currently under src, and
-                                        #< should be removed or renamed
-      $<INSTALL_INTERFACE:include/>
-      $<INSTALL_INTERFACE:include/3rdparty/>
-  )
+add_library(ccf STATIC ${CCF_IMPL_SOURCE})
 
-  target_link_libraries(
-    ccf.snp
-    PUBLIC ${LINK_LIBCXX}
-           -lgcc
-           http_parser.snp
-           ccf_js.snp
-           ccf_endpoints.snp
-           ccfcrypto.snp
-           ccf_kv.snp
-           nghttp2
-           ${CMAKE_THREAD_LIBS_INIT}
-  )
-
-  set_property(TARGET ccf.snp PROPERTY POSITION_INDEPENDENT_CODE ON)
-
-  add_san(ccf.snp)
-
-  if(CCF_DEVEL)
-    install(
-      TARGETS ccf.snp
-      EXPORT ccf
-      DESTINATION lib
-    )
-  endif()
+target_compile_definitions(
+  ccf PUBLIC INSIDE_ENCLAVE VIRTUAL_ENCLAVE _LIBCPP_HAS_THREAD_API_PTHREAD
```

Comment:
I think _LIBCPP_HAS_THREAD_API_PTHREAD was necessary in the libc++/OE days, but probably isn't anymore. Will check separately.

Reply:
It isn't: https://github.com/microsoft/CCF/pull/7111